File: 4-strings/3-string-searching/main.cpp
// String Searching - C++ Example

#include <iostream>
#include <string>

int main() {
    std::string text = "Hello, world!";
    std::string substring = "world";

    size_t index = text.find(substring);
    if (index != std::string::npos) {
        std::cout << "Index of 'world': " << index << std::endl;
    } else {
        std::cout << "'world' not found" << std::endl;
    }
    return 0;
}

File: 4-strings/3-string-searching/main.java
// String Searching - Java Example

class StringSearching {

    public static void main(String[] args) {
        String text = "Hello, world!";
        String substring = "world";

        int index = text.indexOf(substring);
        if (index != -1) {
            System.out.println("Index of 'world': " + index);
        } else {
            System.out.println("'world' not found");
        }
    }
}
